app/views/handlers/block.py

```diff
@@ -78,6 +79,41 @@ def router(self):
             metadata=self._questionnaire_store.metadata,
         )
 
+    @cached_property
+    def rendered_block(self):
+        transformed_block = transform_variants(
+            self.block,
+            self._schema,
+            self._questionnaire_store.metadata,
+            self._questionnaire_store.answer_store,
+            self._questionnaire_store.list_store,
+            self._current_location,
+        )
+        transformed_block_page_title = transformed_block.get("page_title")
+
+        if "question" in transformed_block:
+            question_page_title = transformed_block.get(
+                "page_title"
+            ) or self._get_safe_page_title(transformed_block["question"]["title"])
+
+            self._set_page_title(question_page_title)
+            rendered_question = self.placeholder_renderer.render(
+                transformed_block["question"], self._current_location.list_item_id
+            )
+            return {
+                **transformed_block,
+                **{"question": rendered_question},
+            }
+
+        content_page_title = transformed_block_page_title or self._get_content_title(
+            transformed_block
+        )
+        self._set_page_title(content_page_title)
+        rendered_question = self.placeholder_renderer.render(
+            transformed_block, self._current_location.list_item_id
+        )
+        return rendered_question
```

Comment:
I don't understand why the question and content specific code has been moved here. Could that code stay in the respective handlers, and call the common `_set_page_title` method?

Reply:
Can do, just trying to keep the transform in the block